build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -159,10 +173,62 @@ export function toReportKind(
 			break;
 		}
 
+		case "legacy-compat": {
+			for (const [pkg, details] of Object.entries(report)) {
+				toReturn[pkg] =
+					details.releaseGroup === "client"
+						? details.ranges.legacyCompat
+						: details.ranges.caret;
+			}
+			break;
+		}
+
 		default: {
 			throw new Error(`Unexpected ReportKind: ${kind}`);
 		}
 	}
 
 	return toReturn;
 }
+
+/**
+ * Generates a new semantic version representing the next version in a legacy compatibility range based on a specified multiple of minor versions.
+ *
+ * This function returns the minor version of the given version to the nearest  multiple of `compatVersionInterval` and bumps it by the `compatVersionInterval` to generate
+ * a new semantic version.
+ *
+ * @param version - A semver-compatible string or `semver.SemVer` object representing the current version.
+ * @param compatVersionInterval - The multiple of minor versions to use for calculating the next version in the range.
+ *
+ * @returns A new `semver.SemVer` object representing the next version in the legacy compatibility range.
+ */
+export function getInternalVersionRange(
```

Comment:
I think I misled you with an earlier comment about this function name. I think the poorly named function is the one in version-tools, which would be a separate change. Sorry for being confusing.

Anyway, I think _this_ function, the one that returns a legacy range, should be called `getLegacyCompatVersionRange`.